lib/public/FullTextSearch/Model/IDocumentAccess.php

```diff
@@ -0,0 +1,264 @@
+<?php
+declare(strict_types=1);
+
+
+/**
+ * FullTextSearch - Full text search framework for Nextcloud
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Maxence Lange <maxence@artificial-owl.com>
+ * @copyright 2018
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+
+namespace OCP\FullTextSearch\Model;
+
+
+/**
+ * Interface IDocumentAccess
+ *
+ * This object is used as a data transfer object when
+ *
+ * - indexing a document,
+ * - generating a search request.
+ *
+ * During the index, it is used to define which users, groups, circles, ...
+ * have access to the IIndexDocument
+ *
+ * During the search, it is internally use to define to which group, circles, ...
+ * a user that perform the search belongs to.
+ *
+ * @see IIndexDocument::setAccess
+ *
+ * @since 15.0.0
```

Comment:
Needs to be 16 on all new code
```suggestion
 * @since 16.0.0
```